File: Python/4. 1차원 배열/8958.py
a = int(input())
result = []
for i in range(a):
  cnt = sum = 0
  sen = list(map(str, input()))
  for j in sen:
    if j == 'O':
      cnt += 1
      sum += cnt
    else:
      cnt = 0
  result.append(sum)
  sum = 0
print(*result, sep='\n')

File: C++/4. 1차원 배열/8958.cpp
#include <iostream>

using namespace std;

char arr[81];

int main(){
    int n;
    scanf("%d", &n);

    for(int i=0; i<n; i++){
        int sum=0, cnt=1;
        scanf("%s", arr);

        string s=arr; // char to string
        
        for(int j=0; j<s.size(); j++){
            if(s[j]=='O'){
                sum+=cnt;
                cnt++;
            }
            else cnt=1;
        }

        printf("%d\n", sum);
    }

    return 0;
}